frontend/packages/integration-tests-cypress/support/login.ts

```diff
@@ -1,59 +1,79 @@
-    if (win.SERVER_FLAGS?.authDisabled) {
-      cy.task('log', '  skipping login, console is running with auth disabled');
-      return;
-    }
+// ex: cy.login('my-idp', 'my-user', 'my-password', 'https://my-idp.com')
+Cypress.Commands.add(
+  'login',
+  (
+    provider: string = 'kube:admin',
+    username: string = 'kubeadmin',
+    password: string = Cypress.env('BRIDGE_KUBEADMIN_PASSWORD'),
+    oauthURL: string = Cypress.env('OAUTH_BASE_ADDRESS'),
+  ) => {
 
-    // Make sure we clear the cookie in case a previous test failed to logout.
-    cy.clearCookie('openshift-session-token');
+  cy.session([provider, username], () => {
+    cy.visit('/'); // visits baseUrl which is set in plugins/index.js
+    cy.window().then((win: any) => {
 
-    const idp = provider || KUBEADMIN_IDP;
-    cy.task('log', `  Logging in as ${username || KUBEADMIN_USERNAME}`);
-    cy.byLegacyTestID('login').should('be.visible');
-    cy.get('body').then(($body) => {
-      if ($body.text().includes(idp)) {
-        cy.contains(idp).should('be.visible').click();
+      // Check if auth is disabled (for a local development environment)
+      if (win.SERVER_FLAGS?.authDisabled) {
+        cy.task('log', '  skipping login, console is running with auth disabled');
+        return;
+      }
+
+      if (isLocalDevEnvironment) {
+        cy.origin(oauthURL, { args: { provider, username, password } }, ({ provider, username, password }) => {
+          // note required duplication in else below due to limitations of cy.origin
+          cy.task('log', `  Logging in as ${username}`);
+          cy.get('[data-test-id="login"]').should('be.visible');
+          cy.get('body').then(($body) => {
+            if ($body.text().includes(provider)) {
+              cy.contains(provider).should('be.visible').click();
+            }
+          });
+          cy.get('#inputUsername').type(username);
+          cy.get('#inputPassword').type(password);
+          cy.get('button[type=submit]').click();
+        });
+      } else {
+        // note required duplication in if above due to limitations of cy.origin
+        cy.task('log', `  Logging in as ${username}`);
+        cy.get('[data-test-id="login"]').should('be.visible');
+        cy.get('body').then(($body) => {
+          if ($body.text().includes(provider)) {
+            cy.contains(provider).should('be.visible').click();
+          }
+        });
+        cy.get('#inputUsername').type(username);
+        cy.get('#inputPassword').type(password);
+        cy.get('button[type=submit]').click();
       }
     });
-    cy.get('#inputUsername').type(username || KUBEADMIN_USERNAME);
-    cy.get('#inputPassword').type(password || Cypress.env('BRIDGE_KUBEADMIN_PASSWORD'));
-    cy.get(submitButton).click();
-    masthead.username.shouldBeVisible();
-  });
+
+  }, { cacheAcrossSpecs: true });
+  cy.visit('/'); // visits baseUrl which is set in plugins/index.js
 });
 
-Cypress.Commands.add('logout', () => {
```

Comment:
I opted not to remove it because nearly every existing test is using it (in all packages).  But we certainly can.

Reply:
I worry that someone unfamiliar with these changes or how sessions work would expect a call to `cy.logout()` log out of the console, but it would just be a no-op. I think it would be worth keeping the logout command, but it should actually log the current user out when used. It should be safe to remove its usage in most places now that we're using sessions.

Reply:
Removed and added a standalone logout test.